### DISCUSSION TITLE
Request for Guidance on Access Restoration to OSM Nominatim API

### DISCUSSION THREAD

**Discussion by rsong0606** [osm-search/Nominatim#3420] [Q&A]
Dear OSM Nominatim Team,

I hope this message finds you well. I am writing to seek your assistance regarding an issue I recently encountered with the OSM Nominatim API. I have been receiving a 403 Forbidden error, which I believe is due to unintentional violations of your usage policy.

I acknowledge that I did not fully understand the terms of use before implementing my requests and I apologize for any inconvenience caused. I am committed to complying with all guidelines and would appreciate any guidance on the necessary steps to restore access to the API.

Could you please inform me about the typical duration of the block and any actions I might take to expedite the process of regaining access? Additionally, I would be grateful for any suggestions on how to adjust my usage to prevent future issues.

Thank you for your time and support. I look forward to your guidance and am ready to make the necessary changes to align with your policies!

Heres the attachment.
![WeChatc034d2fb4d3e727738f01360f4f89b52](https://github.com/osm-search/Nominatim/assets/58720071/a8ec49f4-0295-456f-a0cd-6601a4224293)

**Comment by lonvia** [comment]
Can you please be more specific what you were doing with the service? In particular, I'd be interested what library you are using to access the service.

**Reply by rsong0606** [reply]
@lonvia 
Hi! I am using pypostal + a few more pre-processing functions to normalize and parse address into different components. At the last step, call Nominatim to compare if both addresses are matching or not basically it's an address verification project.

In a bit details, after parsing, I will fill in the information into the structured search, by having Street(road), city, state, country & postal code. If both addresses getting output from Nominatim, compare each component and derive the result as a match or mismatch. Hope it is clear now for you to understand. And thanks for helping out!

I am calling the service by open_street_map_url = "https://nominatim.openstreetmap.org/search"

I think I had the code time.sleep(0.5)  triggering this 403 error when doing the experiment.

Here's the attachment from the error.
![WeChatc034d2fb4d3e727738f01360f4f89b52](https://github.com/osm-search/Nominatim/assets/58720071/5fb78c08-f689-47c7-a544-3474fd2c7b0d)

**Comment by lonvia** [comment]
Please set a [custom user agent](https://requests.readthedocs.io/en/latest/user/quickstart/#custom-headers) and use a sleep time of 1s between requests. Also take into consideration that bulk requests are discouraged. Sending a couple of thousand requests once or twice is okay. But if you have a huge amount of addresses or regularly have to send bulk requests, then consider alternatives like setting up your own installation or using a commercial provider.

**Reply by rsong0606** [reply]
@lonvia Thanks for the suggestions. Also may I know how to get myself unblocked? I was browsing online, people mentioned it would be few hours, however, it's been a few days I am still getting the 403 error message.
